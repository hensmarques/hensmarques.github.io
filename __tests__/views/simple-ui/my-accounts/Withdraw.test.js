import React from 'react';
import { Withdraw } from '../../../../src/views/simple-ui/my-accounts/Withdraw';

describe('<Withdraw/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<Withdraw />);
    expect(wrapper).toMatchSnapshot();
  });

  it('updates state when TextInput values change', () => {
    const wrapper = shallow(<Withdraw />);
    expect(wrapper.state()).toEqual({ amount: '', address: '' });
    const mockEventAmount = { target: { value: '5' } };
    const mockEventAddress = { target: { value: '1FCMHSWjy6cWPVoKQ5YM7FgY9C2XuMcMQf' } };
    wrapper.find({ name: 'Amount' }).simulate('change', mockEventAmount);
    wrapper.find({ name: 'Address' }).simulate('change', mockEventAddress);
    expect(wrapper.state()).toEqual({
      amount: mockEventAmount.target.value,
      address: mockEventAddress.target.value
    });
  });

  xit('dispatches makeWithdrawal when form is submitted', () => {
    const makeWithdrawal = jest.fn();
    const mockCurrency = 'BTC';
    const mockAmount = '5';
    const mockAddress = '1FCMHSWjy6cWPVoKQ5YM7FgY9C2XuMcMQf';
    const wrapper = shallow(<Withdraw currency={mockCurrency} makeWithdrawal={makeWithdrawal} />);
    const withdrawSpy = jest.spyOn(wrapper.instance(), 'withdraw');
    wrapper.instance().forceUpdate();
    wrapper.setState({ amount: mockAmount, address: mockAddress });
    wrapper.find('Form').simulate('submit');
    expect(withdrawSpy).toHaveBeenCalled();
    expect(makeWithdrawal).toHaveBeenCalledWith({
      currency_id: mockCurrency,
      amount: mockAmount,
      address: mockAddress
    });
  });
});
