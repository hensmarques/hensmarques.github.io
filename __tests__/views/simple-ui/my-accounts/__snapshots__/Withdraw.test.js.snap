// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Withdraw/> renders and matches the snapshot 1`] = `
<Form
  onSubmit={[Function]}
>
  <TextInput
    className=""
    disabled={true}
    errorMessages={Array []}
    hideInput={false}
    name="Amount"
    onChange={[Function]}
    placeholder=""
    placeholderParam={Object {}}
    type="text"
    value=""
    writeOnce={false}
  />
  <TextInput
    className=""
    disabled={true}
    errorMessages={Array []}
    hideInput={false}
    name="Address"
    onChange={[Function]}
    placeholder=""
    placeholderParam={Object {}}
    type="text"
    value=""
    writeOnce={false}
  />
  <Button
    className=""
    disabled={false}
    loading={false}
    loadingStatus={false}
    onClick={[Function]}
    type="submit"
  >
    Withdraw
  </Button>
</Form>
`;
